### ISSUE TITLE
Upload button does not work for contract ABI

### ISSUE THREAD

**Issue by lana-shanghai** [polkadot-js/apps#4044]
Hello! I'm connected to Canvas testnet. The contract ABI is recognized, but the upload button is grey. Please see attached image
<img width="983" alt="Screenshot 2020-11-18 at 15 14 12" src="https://user-images.githubusercontent.com/31368580/99541453-20badf00-29b1-11eb-89c0-2425d207f3cf.png">

**Comment by jacogr** [comment]
Did you add a `.contract` file or a `.json` (ABI-only) file?

Seems like flipper above?

**Comment by lana-shanghai** [comment]
Well previously there were two separate lines for the .wasm and the .json files, and now it only asks for the ABI. Yes, it's Flipper, I'd like to know how to deploy before I try to deploy anything more complicated :D

**Comment by lana-shanghai** [comment]
OK, now I see the WASM line. Thanks for the help!

**Comment by jacogr** [comment]
Ok, so just clear, re-add, think you ended up with that weird wasm-first upload doesn't-reset issue.

The reason for only the single field currently is that the `.contract` file actually contains both the ABI and the ASM - so it is a single upload when available.

**Comment by jacogr** [comment]
Will leave it open and close with the fix for whatever cause the weirdness.

**Comment by lana-shanghai** [comment]
How do you generate a .contract file? I ran the ```cargo contract generate-metadata``` and it produces only the .wasm and the ABI... No?

**Comment by jacogr** [comment]
It is very new, see the request for support - https://github.com/polkadot-js/apps/issues/4010

**Comment by cepwin** [comment]
I am having a similar error now with the new 2-page upload for contracts.    I went so far to clone the code to my system and run it locally and I still have the error....I go to the first page upload flipper either with .wasm and the .json or .contract...either way when I go to the second page I can not let the deploy button to become active.   Last week with the 1 page deployment it worked just fine. The impression I get from this bug report is it was closed so could I be doing something wrong....if I am I don't see it.
![Screenshot from 2021-02-10 18-35-18](https://user-images.githubusercontent.com/4396376/107586882-cf487780-6bce-11eb-8657-7ec159891c35.png)
![Screenshot from 2021-02-10 18-34-43](https://user-images.githubusercontent.com/4396376/107586927-dd969380-6bce-11eb-826b-821778044558.png)
